controllers/argocd/util.go

```diff
@@ -364,7 +364,33 @@ func (r *ReconcileArgoCD) getDexOAuthClientSecret(cr *argoprojv1a1.ArgoCD) (*str
 	}
 
 	if tokenSecret == nil {
-		return nil, errors.New("unable to locate ServiceAccount token for OAuth client secret")
+		secret := &corev1.Secret{
```

Comment:
Optional: could we have a comment here that mentions that this is done for k8s v1.24? just so we can have the context for it when going through the code in the future?

Not a required change by any means

Reply:
yeah sure , will add comment Thanks @jaideepr97